proposals/2021-01-stargate-upgrade-b/README.md

```diff
@@ -0,0 +1,149 @@
+
+Cosmos Stargate integration success with exchanges and wallet providers reflects the quality of the code developed by the various Cosmos teams.
+
+## Prop 29
+
+This upgrade also implements the fund recovery procedure defined in proposal 29. The code is [here](https://github.com/cosmos/gaia/blob/main/app/prop29.go). As requested during the discussion of the governance proposal, the migration and fund recovery procedure verifies all signatures. This code was tested during the cosmoshub-3 simulated upgrade testnet.
+
+## Stargate Upgrade Steps
+
+------------
+The upgrade steps for the simulated upgrade of the current Cosmoshub mainnet to Cosmos Stargate includes the following steps:
+
+This section is with the current `gaia 2.0.*` implementation.
+
+  1. Validators should set their `gaia` with a halt time of `1613628000`ie. 06:00 UTC on Jan 28th in Unix time.
+
+  2. Validators should then export the current cosmos state with `gaiad export > cosmoshub-3-export.json`
+
+  3. Validators should determine the height of the last block as `last-cosmoshub-3_block`.
+
+  4. Validators should back up their `.gaiad` directory.
+
+This section is with the upgraded `gaia 4.0.*` implemenataion.
+
+  1. Validators should then migrate the exported genesis state. `gaiad migrate cosmoshub-3-export.json --chain-id=cosmoshub-4 --initial-height [last_cosmoshub-3_block+1] > genesis.json`
+  2. Validators should (**AFTER ENSURING IT IS BACKED UP**!) delete their `~/.gaiad` directory and create a new one with `gaiad init [name]` and then edit their configuration files or recover the config files from backups.
+  3. Validators should then start `cosmoshub-4` with `gaiad start`. Automated audits of the genesis state can take 30-120 min using the crisis module. This can be disabled by `gaiad start --x-crisis-skip-assert-invariants`.
+
+Validators should expect that at least 16GB of RAM needs to be provisioned to process the first new block on cosmoshub-4.
+
+## Time of the Upgrade
+
+------------
+We propose scheduling the Cosmoshub-3 to Cosmoshub-4 upgrade for Thursday Feb 18th, 2021 at 0600 UTC
+
+- West Coast USA: 10 PM on Feb 17
+- East Coast USA: 1 AM on Feb 18
+- Central Europe: 7 AM on Feb 18
+- Seoul: 5 PM on Feb 18
+
+## The git commit of Gaia that we are upgrading to
+
+------------
+`2bb04266266586468271c4ab322367acbf41188f`
+
+## What we're doing while voting is happening
+
+------------
+
+We expect that integration partners will be motivated to finish their preparations for the upgrade during the period when the governance proposal is running.
+
+A testnet with the final version of Gaia 3.0 will be running to assist with integrations.
+
+[Stargate Ecosystem Readiness Report](https://github.com/cosmos/stargate-launch/blob/master/ecosystem_readiness.md) is a living document. We will be updating this document as we get reports of completed end to end testing. We expect a hard deadline to be motivating.
+
+## What is the upgrade abort process
```

Comment:
Does it make sense to update this section with the process that was followed for the recent v3 upgrade proposal? i.e., signal the cancellation of the upgrade proposal with a follow-on proposal, followed by a new upgrade proposal?